sale_salesperson_filter/models/sale_order.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        super(SaleOrder, self).onchange_partner_id()
+        self.update({
+            'user_id': self.partner_id.user_id,
+        })
```

Comment:
I don't think this method is needed (standard code already covers the logic).  Besides, it will probably make Odoo throw a server error if partner_id is blank.